packages/indexer-agent/src/agent.ts

```diff
@@ -116,147 +124,244 @@ export const convertSubgraphBasedRulesToDeploymentBased = (
+    networks.every(
+      (network, index) =>
+        network.specification.networkIdentifier ===
+        operators[index].specification.networkIdentifier,
+    )
+  if (!validInputs) {
+    throw new Error(
+      'Invalid Networks and Operator pairs used in Agent initialization',
+    )
+  }
+  // Note on undefineds: `lodash.zip` can return `undefined` if array lengths are
+  // uneven, but we have just checked that.
+  const networksAndOperators = zip(networks, operators).map(pair => {
+    const [network, operator] = pair
+    return { network: network!, operator: operator! }
+  })
+  return new MultiNetworks(networksAndOperators, networkAndOperatorIdentity)
+}
+
+export class Agent {
   logger: Logger
   metrics: Metrics
-  indexer: Indexer
-  network: Network
-  networkMonitor: NetworkMonitor
-  networkSubgraph: NetworkSubgraph
-  allocateOnNetworkSubgraph: boolean
-  registerIndexer: boolean
+  graphNode: GraphNode
+  multiNetworks: MultiNetworks<NetworkAndOperator>
+  indexerManagement: IndexerManagementClient
   offchainSubgraphs: SubgraphDeploymentID[]
-  receiptCollector: ReceiptCollector
 
   constructor(
     logger: Logger,
     metrics: Metrics,
-    indexer: Indexer,
-    network: Network,
-    networkMonitor: NetworkMonitor,
-    networkSubgraph: NetworkSubgraph,
-    allocateOnNetworkSubgraph: boolean,
-    registerIndexer: boolean,
+    graphNode: GraphNode,
+    operators: Operator[],
+    indexerManagement: IndexerManagementClient,
+    networks: Network[],
     offchainSubgraphs: SubgraphDeploymentID[],
-    receiptCollector: ReceiptCollector,
   ) {
     this.logger = logger.child({ component: 'Agent' })
     this.metrics = metrics
-    this.indexer = indexer
-    this.network = network
-    this.networkMonitor = networkMonitor
-    this.networkSubgraph = networkSubgraph
-    this.allocateOnNetworkSubgraph = allocateOnNetworkSubgraph
-    this.registerIndexer = registerIndexer
+    this.graphNode = graphNode
+    this.indexerManagement = indexerManagement
+    this.multiNetworks = createMultiNetworks(networks, operators)
     this.offchainSubgraphs = offchainSubgraphs
-    this.receiptCollector = receiptCollector
   }
 
   async start(): Promise<Agent> {
+    // --------------------------------------------------------------------------------
+    // * Connect to Graph Node
+    // --------------------------------------------------------------------------------
     this.logger.info(`Connect to Graph node(s)`)
-    await this.indexer.connect()
+    await this.graphNode.connect()
     this.logger.info(`Connected to Graph node(s)`)
 
-    // Ensure there is a 'global' indexing rule
-    await this.indexer.ensureGlobalIndexingRule()
+    // --------------------------------------------------------------------------------
+    // * Ensure there is a 'global' indexing rule
+    // --------------------------------------------------------------------------------
+    await this.multiNetworks.map(({ operator }) =>
```

Comment:
Yes! Nice catch!
I was rewriting this from the previous version of the code, part by part, and didn't notice it could be wrapped into a single operation

Reply:
Fixed in 1c006ad7